test/framework/db/acl_test.clj

```diff
@@ -0,0 +1,515 @@
+                        :name    "John"
+                        :surname "Doe"
+                        :email   "doe.john@test.com"
+                        :role    :customer}
+                       {:id      2
+                        :name    "Admin"
+                        :surname "Doe"
+                        :email   "doe.admin@test.com"
+                        :role    :administrator}
+                       {:id      3
+                        :name    "Warehouse"
+                        :surname "Boy"
+                        :email   "boy@warehouse.com"
+                        :role    :warehouse-worker}
+                       {:id      4
+                        :name    "Shop"
+                        :surname "Bi"
+                        :email   "bishop@warehouse.com"
+                        :role    :shop-worker}
+                       {:id      5
+                        :name    "Postal"
+                        :surname "Service"
+                        :email   "postoffice@warehouse.com"
+                        :role    :postal-worker}
+                       {:id      6
+                        :name    "Janet"
+                        :surname "Doe"
+                        :email   "doe.janet@test.com"
+                        :role    :customer}]
+   :items             [{:id    1
+                        :name  "salami slicer"
+                        :price {:value    28.49
+                                :currency "USD"}}
+                       {:id    2
+                        :name  "bread knife"
+                        :price {:value    12.97
+                                :currency "USD"}}
+                       {:id    3
+                        :name  "Meat tenderizer"
+                        :price {:value    14.99
+                                :currency "USD"}}
+                       {:id    4
+                        :name  "cheese grater"
+                        :price {:value    12.49
+                                :currency "USD"}}]
+   :addresses         [{:id          1
+                        :country     "Czech Republic"
+                        :city        "Prague"
+                        :street      "Apolinářská"
+                        :number      "447/4A"
+                        :postal-code "128 00"}
+                       {:id          2
+                        :country     "Hungary"
+                        :city        "Budapest"
+                        :street      "Teréz krt."
+                        :number      "33"
+                        :postal-code "1083"}]
+   :postal-addresses  [{:user-id    1
+                        :address-id 1}
+                       {:user-id    2
+                        :address-id 2}]
+   :invoice-addresses [{:user-id    1
+                        :address-id 1}
+                       {:user-id    2
+                        :address-id 2}]
+   :roles             [{:customer [{:table   "items"
```

Comment:
I suggest renaming `:roles` -> `:permissions`
semantics of the value is permissions of particular roles

The same with output of `(add-user-by-id {} 1))`
```
{:session {:user {:id 1,
                  :name "John",
                  :surname "Doe",
                  :email "doe.john@test.com",
                  :role :customer,
                  :roles [{:table "items", :actions [:select], :filter :all}
                          {:table "users", :actions [:select :update :delete], :filter :own}
                          {:table "addresses", :actions [:select :update :delete], :filter :own}
                          {:table "carts", :actions [:select :update :delete], :filter :own}]}}}
```

Reply:
I was in trouble with the name, thanks for the advice!